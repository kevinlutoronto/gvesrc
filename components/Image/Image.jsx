// @flow
import React from 'react';
import cn from 'classnames';
import styles from './style.scss';

import { getDataAttrs } from '../../dataUtils';
import type { Data, StyleAttributes } from '../../types';

const defaultImage = 'https://cdn.spacetelescope.org/archives/images/screen/s82e5407.jpg';

type Props = {|
  style?: StyleAttributes,
  imageStyle?: StyleAttributes,
  src: string,
  altText?: string,
  data?: Data,
  onClick?: (evt: MouseEvent) => void,
  className?: string,
  fallback?: string,
  onLoad?: (evt: MouseEvent) => void,
|};

type State = {|
  errored: boolean,
|};

class Image extends React.Component<Props, State> {
  static defaultProps = {
    style: { width: '200px', height: 'auto' },
    src: defaultImage,
    fallback: defaultImage,
    altText: 'Image',
  };

  state = {
    errored: false,
  };

  getImageUrl() {
    const { src, fallback } = this.props;
    if (this.state.errored || !src) {
      return fallback;
    }
    return src;
  }

  handleError = () => {
    this.setState({ errored: true });
  };

  render() {
    const { className, data, style, altText, imageStyle, onClick, onLoad } = this.props;
    const imageCn = cn(styles.image, className);
    return (
      <div {...getDataAttrs(data)} style={style}>
        <img
          alt={altText}
          className={imageCn}
          src={this.getImageUrl()}
          style={imageStyle}
          onError={this.handleError}
          onClick={onClick}
          onLoad={onLoad}
        />
      </div>
    );
  }
}

Image.displayName = 'Plasma@Image';

export default Image;
