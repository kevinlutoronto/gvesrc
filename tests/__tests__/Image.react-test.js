/* eslint-disable */

import React from 'react';
import Image from '../../components/Image/Image';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

describe('Image Component', () => {
  test('render', () => {
    const component = renderer.create(<Image />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
