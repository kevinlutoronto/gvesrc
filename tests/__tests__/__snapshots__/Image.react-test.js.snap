// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component render 1`] = `
<div
  style={
    Object {
      "height": "auto",
      "width": "200px",
    }
  }
>
  <img
    alt="Image"
    className="image"
    onClick={undefined}
    onError={[Function]}
    onLoad={undefined}
    src="https://cdn.spacetelescope.org/archives/images/screen/s82e5407.jpg"
    style={undefined}
  />
</div>
`;
